### ISSUE TITLE
going in the right direction?

### ISSUE THREAD

**Issue by shaunlebron** [yogthos/clojure-error-message-catalog#14]
I think these error docs might be moving in the wrong direction:

- [clojure-lang-persistentlist-cannot-be-cast-to-clojure-lang-symbol](https://github.com/yogthos/clojure-error-message-catalog/blob/master/clj/clojure-lang-persistentlist-cannot-be-cast-to-clojure-lang-symbol.md)
- [dont-know-how-to-create-iseq-from](https://github.com/yogthos/clojure-error-message-catalog/blob/master/clj/dont-know-how-to-create-iseq-from.md)
- [java-lang-string-cannot-be-cast-to-clojure-lang-ifn](https://github.com/yogthos/clojure-error-message-catalog/blob/master/clj/java-lang-string-cannot-be-cast-to-clojure-lang-ifn.md)
- [nth-not-supported-on-this-type](https://github.com/yogthos/clojure-error-message-catalog/blob/master/clj/nth-not-supported-on-this-type.md)

Some errors can happen in a multitude of contexts, but the above errors attribute them to specific cases only, though they obviously can show up elsewhere.  Sorry, this was just my gut reaction, and I haven't thought about it too much yet, but I thought others might be thinking the same thing, so I wanted to bring it up early.

My secondary reaction is that following in this direction could amount to us documenting error reports for every function/macro arg type.  Surely there is a better plan for dealing with this class of errors?

**Comment by yogthos** [comment]
I agree. It's likely better to name the files after the error message, and then have all the different causes listed for the error. Also, I can add you to the project if you'd like that way you can change things easier. :)
